From b534d8154d1aad153453ea3be030a2397f84710c Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:16:37 -0400
Subject: [alsa] hda/conexant: Enable stereo mix input for CX20549 and CX20551

Message-id: <140972823573-75-patchset-jkysela@redhat.com>
Patchwork-id: 90775
O-Subject: [RHEL 7.1 PATCH 074/442] ALSA: hda/conexant - Enable stereo mix input for CX20549 and CX20551
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Both CX20549 and CX20551 codecs have a mixer widget and it can be
connected as the ADC source.  Like AD and VIA codecs, enable the
add_stereo_mix_input flag for these codecs.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: d50ce6c0b8afb958316fba15e329e3e53abfd968)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_conexant.c b/sound/pci/hda/patch_conexant.c
index 6b2c11d..cf2ee7f 100644
--- a/sound/pci/hda/patch_conexant.c
+++ b/sound/pci/hda/patch_conexant.c
@@ -3424,12 +3424,14 @@ static int patch_conexant_auto(struct hda_codec *codec)
  case 0x14f15045:
   codec->single_adc_amp = 1;
   spec->gen.mixer_nid = 0x17;
+  spec->gen.add_stereo_mix_input = 1;
   snd_hda_pick_fixup(codec, cxt5045_fixup_models,
        cxt5045_fixups, cxt_fixups);
   break;
  case 0x14f15047:
   codec->pin_amp_workaround = 1;
   spec->gen.mixer_nid = 0x19;
+  spec->gen.add_stereo_mix_input = 1;
   snd_hda_pick_fixup(codec, cxt5047_fixup_models,
        cxt5047_fixups, cxt_fixups);
   break;
-- 
1.7.1